### ISSUE TITLE
Scanning progress bar is not closed

### ISSUE THREAD

**Issue by adamshakhabov** [yuenm18/ooxml-viewer-vscode#26]
The Viewer sometimes doesn't close the progress bar after scanning. Maybe some error occurred, but the result is good, and I keep work. It is just a little noise.🙂 Only VS Code re-openning helps.

![Code_zP6jFTHeTV](https://user-images.githubusercontent.com/5969304/150594610-f0cd0ce5-00a6-41b2-baf9-a331137ad2a1.gif)

For example, today it happened with this file: [**close-bar-error.pptx**](https://github.com/yuenm18/ooxml-viewer-vscode/files/7916375/close-bar-error.pptx)

Not sure is it relevant, but I collected some logs that VS Code provides:

```
[2022-01-21 11:42:28.239] [exthost] [info] Extension host with pid 17780 started
[2022-01-21 11:42:28.239] [exthost] [info] Skipping acquiring lock for c:\Users\thead\AppData\Roaming\Code\User\workspaceStorage\13cb7a9017e7b13098adc60ac4b9c66a.
[2022-01-21 11:42:28.518] [exthost] [info] ExtensionService#_doActivateExtension vscode.microsoft-authentication, startup: false, activationEvent: 'onAuthenticationRequest:microsoft'
[2022-01-21 11:42:28.538] [exthost] [info] ExtensionService#_doActivateExtension vscode.html-language-features, startup: false, activationEvent: 'onLanguage:html'
[2022-01-21 11:42:28.560] [exthost] [info] ExtensionService#_doActivateExtension yuenm18.ooxml-viewer, startup: false, activationEvent: 'onView:ooxmlViewer'
[2022-01-21 11:42:28.669] [exthost] [info] ExtensionService#_doActivateExtension vscode.debug-auto-launch, startup: true, activationEvent: '*'
[2022-01-21 11:42:28.679] [exthost] [info] ExtensionService#_doActivateExtension vscode.git-base, startup: true, activationEvent: '*', root cause: vscode.github
[2022-01-21 11:42:28.688] [exthost] [info] ExtensionService#_doActivateExtension donjayamanne.githistory, startup: true, activationEvent: '*'
[2022-01-21 11:42:28.842] [exthost] [info] ExtensionService#_doActivateExtension redhat.vscode-yaml, startup: true, activationEvent: '*', root cause: fbaligand.vscode-logstash-editor
[2022-01-21 11:42:29.115] [exthost] [info] ExtensionService#_doActivateExtension mhutchie.git-graph, startup: true, activationEvent: '*'
[2022-01-21 11:42:29.615] [exthost] [info] ExtensionService#_doActivateExtension vscode.git, startup: false, activationEvent: 'api', root cause: donjayamanne.githistory
[2022-01-21 11:42:29.741] [exthost] [info] ExtensionService#_doActivateExtension vscode.github, startup: true, activationEvent: '*'
[2022-01-21 11:42:29.751] [exthost] [info] ExtensionService#_doActivateExtension fbaligand.vscode-logstash-editor, startup: true, activationEvent: '*'
[2022-01-21 11:42:30.479] [exthost] [info] Eager extensions activated
[2022-01-21 11:42:30.493] [exthost] [warning] [redhat.vscode-yaml] Accessing a resource scoped configuration without providing a resource is not expected. To get the effective value for '[yaml]', provide the URI of a resource or 'null' for any resource.
[2022-01-21 11:42:30.527] [exthost] [info] ExtensionService#_doActivateExtension vscode.github-authentication, startup: false, activationEvent: 'onAuthenticationRequest:github'
[2022-01-21 11:42:30.544] [exthost] [info] ExtensionService#_doActivateExtension vscode.emmet, startup: false, activationEvent: 'onStartupFinished'
[2022-01-21 11:42:30.556] [exthost] [info] ExtensionService#_doActivateExtension vscode.merge-conflict, startup: false, activationEvent: 'onStartupFinished'
[2022-01-21 11:42:30.560] [exthost] [info] ExtensionService#_doActivateExtension ms-vscode-remote.remote-wsl-recommender, startup: false, activationEvent: 'onStartupFinished'
[2022-01-21 11:42:30.572] [exthost] [info] ExtensionService#_doActivateExtension alefragnani.Bookmarks, startup: false, activationEvent: 'onStartupFinished'
[2022-01-21 11:42:30.609] [exthost] [info] ExtensionService#_doActivateExtension eamodio.gitlens, startup: false, activationEvent: 'onStartupFinished'
[2022-01-21 11:42:31.949] [exthost] [info] ExtensionService#_doActivateExtension vscode.extension-editing, startup: false, activationEvent: 'onLanguage:markdown'
[2022-01-21 11:42:31.964] [exthost] [info] ExtensionService#_doActivateExtension vscode.markdown-language-features, startup: false, activationEvent: 'onLanguage:markdown'
[2022-01-21 11:42:32.260] [exthost] [info] ExtensionService#_doActivateExtension vscode.markdown-math, startup: false, activationEvent: 'api', root cause: vscode.markdown-language-features
[2022-01-21 11:42:32.272] [exthost] [info] ExtensionService#_doActivateExtension bierner.markdown-checkbox, startup: false, activationEvent: 'api', root cause: vscode.markdown-language-features
[2022-01-21 11:42:32.291] [exthost] [info] ExtensionService#_doActivateExtension bierner.markdown-emoji, startup: false, activationEvent: 'api', root cause: vscode.markdown-language-features
[2022-01-21 11:42:32.315] [exthost] [info] ExtensionService#_doActivateExtension bierner.markdown-mermaid, startup: false, activationEvent: 'api', root cause: vscode.markdown-language-features
[2022-01-21 11:42:32.334] [exthost] [info] ExtensionService#_doActivateExtension bierner.markdown-preview-github-styles, startup: false, activationEvent: 'api', root cause: vscode.markdown-language-features
[2022-01-21 11:42:32.349] [exthost] [info] ExtensionService#_doActivateExtension bierner.markdown-yaml-preamble, startup: false, activationEvent: 'api', root cause: vscode.markdown-language-features
[2022-01-21 11:42:32.664] [exthost] [info] ExtensionService#_doActivateExtension vscode.configuration-editing, startup: false, activationEvent: 'onLanguage:jsonc'
[2022-01-21 11:42:32.675] [exthost] [info] ExtensionService#_doActivateExtension vscode.json-language-features, startup: false, activationEvent: 'onLanguage:jsonc'
[2022-01-21 11:42:32.704] [exthost] [info] ExtensionService#_doActivateExtension vscode.typescript-language-features, startup: false, activationEvent: 'onLanguage:jsonc'
[2022-01-21 15:29:44.955] [exthost] [info] ExtensionService#_doActivateExtension redhat.vscode-xml, startup: false, activationEvent: 'onLanguage:xml'
[2022-01-21 15:29:48.050] [exthost] [warning] [redhat.vscode-xml] Accessing a resource scoped configuration without providing a resource is not expected. To get the effective value for '[xml]', provide the URI of a resource or 'null' for any resource.
[2022-01-21 15:42:48.305] [exthost] [error] [redhat.vscode-xml] provider FAILED
[2022-01-21 15:42:48.307] [exthost] [error] Canceled: Canceled
	at S.makeCancelError (c:\Users\thead\.vscode\extensions\redhat.vscode-xml-0.18.2\dist\extension.js:2:362236)
	at S.handleFailedRequest (c:\Users\thead\.vscode\extensions\redhat.vscode-xml-0.18.2\dist\extension.js:2:362087)
	at c:\Users\thead\.vscode\extensions\redhat.vscode-xml-0.18.2\dist\extension.js:2:326960
	at async s.provideDocumentSymbols (c:\Users\thead\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:95:87704)
[2022-01-21 21:56:24.504] [exthost] [info] extension host terminating: renderer closed the socket
[2022-01-21 21:56:24.596] [exthost] [info] Extension host with pid 17780 exiting with code 0
[2022-01-21 21:56:26.824] [exthost] [info] Extension host with pid 15336 started
[2022-01-21 21:56:26.824] [exthost] [info] Skipping acquiring lock for c:\Users\thead\AppData\Roaming\Code\User\workspaceStorage\0bfe12cb68c80a72ddf819aae4eaf799.
[2022-01-21 21:56:27.319] [exthost] [info] ExtensionService#_doActivateExtension vscode.microsoft-authentication, startup: false, activationEvent: 'onAuthenticationRequest:microsoft'
[2022-01-21 21:56:27.338] [exthost] [info] ExtensionService#_doActivateExtension vscode.extension-editing, startup: false, activationEvent: 'onLanguage:markdown'
[2022-01-21 21:56:27.352] [exthost] [info] ExtensionService#_doActivateExtension vscode.markdown-language-features, startup: false, activationEvent: 'onLanguage:markdown'
[2022-01-21 21:56:27.383] [exthost] [info] ExtensionService#_doActivateExtension yuenm18.ooxml-viewer, startup: false, activationEvent: 'onView:ooxmlViewer'
[2022-01-21 21:56:27.510] [exthost] [info] ExtensionService#_doActivateExtension vscode.debug-auto-launch, startup: true, activationEvent: '*'
[2022-01-21 21:56:27.521] [exthost] [info] ExtensionService#_doActivateExtension vscode.git-base, startup: true, activationEvent: '*', root cause: vscode.github
[2022-01-21 21:56:27.535] [exthost] [info] ExtensionService#_doActivateExtension donjayamanne.githistory, startup: true, activationEvent: '*'
[2022-01-21 21:56:27.682] [exthost] [info] ExtensionService#_doActivateExtension redhat.vscode-yaml, startup: true, activationEvent: '*', root cause: fbaligand.vscode-logstash-editor
[2022-01-21 21:56:27.895] [exthost] [info] ExtensionService#_doActivateExtension mhutchie.git-graph, startup: true, activationEvent: '*'
[2022-01-21 21:56:29.010] [exthost] [info] ExtensionService#_doActivateExtension vscode.git, startup: false, activationEvent: 'api', root cause: donjayamanne.githistory
[2022-01-21 21:56:29.086] [exthost] [info] ExtensionService#_doActivateExtension vscode.markdown-math, startup: false, activationEvent: 'api', root cause: vscode.markdown-language-features
[2022-01-21 21:56:29.277] [exthost] [info] ExtensionService#_doActivateExtension bierner.markdown-checkbox, startup: false, activationEvent: 'api', root cause: vscode.markdown-language-features
[2022-01-21 21:56:29.370] [exthost] [info] ExtensionService#_doActivateExtension bierner.markdown-emoji, startup: false, activationEvent: 'api', root cause: vscode.markdown-language-features
[2022-01-21 21:56:29.436] [exthost] [info] ExtensionService#_doActivateExtension bierner.markdown-mermaid, startup: false, activationEvent: 'api', root cause: vscode.markdown-language-features
[2022-01-21 21:56:29.457] [exthost] [info] ExtensionService#_doActivateExtension bierner.markdown-preview-github-styles, startup: false, activationEvent: 'api', root cause: vscode.markdown-language-features
[2022-01-21 21:56:29.472] [exthost] [info] ExtensionService#_doActivateExtension bierner.markdown-yaml-preamble, startup: false, activationEvent: 'api', root cause: vscode.markdown-language-features
[2022-01-21 21:56:29.594] [exthost] [info] ExtensionService#_doActivateExtension vscode.github, startup: true, activationEvent: '*'
[2022-01-21 21:56:29.606] [exthost] [info] ExtensionService#_doActivateExtension fbaligand.vscode-logstash-editor, startup: true, activationEvent: '*'
[2022-01-21 21:56:30.171] [exthost] [info] Eager extensions activated
[2022-01-21 21:56:30.240] [exthost] [warning] [redhat.vscode-yaml] Accessing a resource scoped configuration without providing a resource is not expected. To get the effective value for '[yaml]', provide the URI of a resource or 'null' for any resource.
[2022-01-21 21:56:30.272] [exthost] [info] ExtensionService#_doActivateExtension vscode.github-authentication, startup: false, activationEvent: 'onAuthenticationRequest:github'
[2022-01-21 21:56:30.292] [exthost] [info] ExtensionService#_doActivateExtension vscode.emmet, startup: false, activationEvent: 'onStartupFinished'
[2022-01-21 21:56:30.305] [exthost] [info] ExtensionService#_doActivateExtension vscode.merge-conflict, startup: false, activationEvent: 'onStartupFinished'
[2022-01-21 21:56:30.312] [exthost] [info] ExtensionService#_doActivateExtension ms-vscode-remote.remote-wsl-recommender, startup: false, activationEvent: 'onStartupFinished'
[2022-01-21 21:56:30.326] [exthost] [info] ExtensionService#_doActivateExtension alefragnani.Bookmarks, startup: false, activationEvent: 'onStartupFinished'
[2022-01-21 21:56:30.479] [exthost] [info] ExtensionService#_doActivateExtension eamodio.gitlens, startup: false, activationEvent: 'onStartupFinished'
[2022-01-21 21:56:32.993] [exthost] [info] ExtensionService#_doActivateExtension redhat.vscode-xml, startup: false, activationEvent: 'onLanguage:xml'
[2022-01-21 21:56:34.289] [exthost] [warning] [redhat.vscode-xml] Accessing a resource scoped configuration without providing a resource is not expected. To get the effective value for '[xml]', provide the URI of a resource or 'null' for any resource.
```

**VS Code info**
Version: 1.63.2 (user setup)
Commit: 899d46d82c4c95423fb7e10e68eba52050e30ba3
Date: 2021-12-15T09:40:02.816Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19044

**Comment by yuenm18** [comment]
I'm having a little trouble reproducing this.  Based on the video you sent it looks like you got the error when saving the powerpoint file.  Does this happen when you modify the xml parts from vscode also or just when you save the file externally? 

How often does this happen?

**Comment by adamshakhabov** [comment]
It doesn't matter _ready_ or _write_. I work a lot of with .pptx and .docx. At the most of the time, this happens with .pptx.

I will try to catch up more repro details and will share here.

**Comment by mikeebowen** [comment]
@adamshakhabov 

Does the number of parts make a difference? E.G. if a .pptx has a lot of slides vs only one does the issue happen more often?

**Comment by adamshakhabov** [comment]
I don't remember exactly😶

**Comment by yuenm18** [comment]
@adamshakhabov Is this still happening? It might have been fixed with #27

**Comment by adamshakhabov** [comment]
No, it does not happen. Looks like some auto-fix.
